.travis.yml

```diff
@@ -1,38 +1,47 @@
-language: python
+language: c
 
-python:
-    - 2.7
+os:
+    - osx
+    - linux
 
-env: DISPLAY=:99 MOZ_XVFB=1
+compiler:
+    - gcc
 
 before_install:
-    - sudo apt-get update
-    - "/sbin/start-stop-daemon --start --quiet --make-pidfile --pidfile /tmp/custom_xvfb_99.pid --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1024x768x24"
+    # set up linux only items
+    - ./.travis/preinstall.sh
 
-install:
-    - sudo apt-get install flashplugin-nonfree subversion
+    - sudo -H pip install virtualenv
+    - virtualenv ~/.venv
+    - source ~/.venv/bin/activate
 
-    # As long as we have invasive changes lets get the trunk version of marionette-client
+install:
+    # As long as we have invasive changes, lets get the trunk version of marionette-client
     - svn checkout https://github.com/mozilla/gecko-dev/trunk/testing/marionette/client
-    - "cd client && python setup.py develop && cd .."
+    - "cd client && sudo python setup.py develop && cd .."
 
-    - python setup.py develop
+    - sudo python setup.py develop
```

Comment:
Why sudo in both of the above cases? I don't see why we cannot install all of our packages in the newly created venv.